tomb-common/src/crypto/rsa.rs

```diff
@@ -0,0 +1,271 @@
+
+#[derive(Debug, Clone)]
+pub struct RsaPrivateKey(rsa::RsaPrivateKey);
+
+//--------------------------------------------------------------------------------------------------
+// Implementations
+//--------------------------------------------------------------------------------------------------
+
+impl RsaPublicKey {
+    /// Gets the public key modulus.
+    pub fn get_public_key_modulus(&self) -> Result<Vec<u8>> {
+        Ok(self.0.n().to_bytes_le())
+    }
+
+    /// Gets base64 string of the SHA-256 fingerprint of the public key's SPKI.
+    pub fn get_fingerprint(&self) -> Result<String> {
+        let document = self.0.to_public_key_der()?;
+        let spki = SubjectPublicKeyInfoOwned::try_from(document.as_bytes())?;
+        let fingerprint = spki.fingerprint_base64()?;
+        Ok(fingerprint)
+    }
+
+    /// Writes the public key to a SPKI PEM file.
+    /// # Arguments
+    /// path - The path to the file to write to.
+    pub fn to_pem_file(&self, path: impl AsRef<std::path::Path>) -> Result<()> {
+        self.0
+            .write_public_key_pem_file(path, LineEnding::LF)
+            .map_err(|e| anyhow!(RsaError::ExportToPemFileFailed(anyhow!(e))))
+    }
+
+    /// Reads the public key from a SPKI PEM file.
+    /// # Arguments
+    /// path - The path to the file to read from.
+    pub fn from_pem_file(path: impl AsRef<std::path::Path>) -> Result<Self> {
+        let key = rsa::RsaPublicKey::read_public_key_pem_file(path)?;
+        Ok(Self(key))
+    }
+
+    /// Read the public key from DER bytes.
+    /// # Arguments
+    /// bytes - The DER bytes to read from.
+    pub fn from_der(bytes: &[u8]) -> Result<Self> {
+        let key = rsa::RsaPublicKey::from_public_key_der(bytes)?;
+        Ok(Self(key))
+    }
+}
+
+// #[cfg(test)]
+impl RsaPrivateKey {
+    /// Constructs a new 2048-bit RSA private key.
+    pub fn new() -> Result<Self> {
+        Ok(Self(rsa::RsaPrivateKey::new(
+            &mut rand_core::OsRng,
+            RSA_KEY_SIZE,
+        )?))
+    }
+
+    /// Writes the private key to a PKCS#8 PEM file.
+    ///
+    /// # Arguments
+    /// path - The path to the file to write to.
+    pub fn to_pem_file(&self, path: impl AsRef<std::path::Path>) -> Result<()> {
+        self.0
+            .write_pkcs8_pem_file(path, LineEnding::LF)
+            .map_err(|e| anyhow!(RsaError::ExportToPemFileFailed(anyhow!(e))))
+    }
+
+    /// Reads the private key from a PKCS#8 PEM file.
+    ///
+    /// # Arguments
+    /// path - The path to the file to read from.
+    pub fn from_pem_file(path: impl AsRef<std::path::Path>) -> Result<Self> {
+        let key = rsa::RsaPrivateKey::read_pkcs8_pem_file(path)?;
+        Ok(Self(key))
+    }
+
+    /// Reads the private key from DER bytes.
+    ///
+    /// # Arguments
+    /// bytes - The DER bytes to read from.
+    pub fn from_der(bytes: &[u8]) -> Result<Self> {
+        let key = rsa::RsaPrivateKey::from_pkcs8_der(bytes)?;
+        Ok(Self(key))
+    }
+
```

Comment:
id like that there be a `to_der()` function exposed here as well

Reply:
I can implement this -- but I am wondering what use case there is for exporting key bytes to der
- this isn't a key you share
- there's already an export method (to_pem_file)